src/parser/mod.rs

```diff
@@ -7506,7 +7506,7 @@ impl<'a> Parser<'a> {
                 with_offset,
                 with_offset_alias,
             })
-        } else if self.parse_keyword(Keyword::JSON_TABLE) {
+        } else if dialect_of!(self is MySqlDialect | AnsiDialect) && self.parse_keyword(Keyword::JSON_TABLE) {
```

Comment:
I suppose `AnsiDialect` is for SQL standard, though it doesn't specify which standard it follows. Seems it makes sense to add it in supported dialect.

Reply:
It could be the default behavior for dialects which are likely to follow the standard to support the table function.

Reply:
As there are only very few dialects supporting this, I think it makes more sense to not add it into `GenericDialect` currently.